### PR TITLE
Replace trace id with local root span id

### PR DESCRIPTION
**Pull request by mar-kolya**
## Description

TraceId may identify multiple requests that come to the same service from the same trace. This means that if we try to look at the profile filtered by the trace id we may see somewhere between 1 and all requests - which is confusing.

## Checklist
- [ ] Added to the correct milestone.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

File: releasenotes/notes/profiling-local-root-span-fd7ee6a08b8612b7.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+fixes:
```
Comment (brettlangdon) [inline releasenotes/notes/profiling-local-root-span-fd7ee6a08b8612b7.yaml:2]: this is a bug fix right?

Comment (brettlangdon) [reply releasenotes/notes/profiling-local-root-span-fd7ee6a08b8612b7.yaml:2]: what changes in the UI/public facing from this change?

just now wondering if this actually needs a changelog entry or not.

File: tests/profiling/collector/test_stack.py [hunk 2]
```diff
@@ -455,7 +455,7 @@ def test_collect_span_id(tracer_and_collector):
         except IndexError:
             # No event left or no event yet
             continue
-        if span.trace_id == event.trace_id and span.span_id == event.span_id:
+        if span.span_id == event.span_id:
```
Comment (brettlangdon) [inline tests/profiling/collector/test_stack.py:458]: do we need to assert the local root span id as well?

Comment (mar-kolya) [reply tests/profiling/collector/test_stack.py:458]: I have a feeling the way this test is setup there is no local root span...

Comment (mar-kolya) [reply tests/profiling/collector/test_stack.py:458]: Added assert for local root span as well

**Review by brettlangdon — APPROVED** [review]

**Review by Kyle-Verhoog — COMMENTED** [review]
Could we elaborate a bit on "Trace id may be confusing for profiling purposes"? I don't really know why the change is needed or desired 🤔 

Otherwise, the code change looks good! :+1:

**Comment by mar-kolya** [comment]
TraceId may identify multiple requests that come to the same service from the same trace. This means that if we try to look at the profile filtered by the trace id we may see somewhere between 1 and all requests - which is confusing.

**Review by Kyle-Verhoog — APPROVED** [review]
thanks!

**Pull request merged**